src/anyp/PortCfg.h

```diff
@@ -36,8 +36,14 @@ class PortCfg : public CodeContext
     virtual ScopedId codeContextGist() const override;
     virtual std::ostream &detailCodeContext(std::ostream &os) const override;
 
+    void print(std::ostream &) const;
+
+    /// the protocol name associated with this port type
+    const char *defaultProtocolName() const;
+
     PortCfgPointer next;
 
+    SBuf directive; ///< the corresponding configuration directive name
```

Comment:
If some of those use cases are better handled using PortCfg::print() instead of just the directive name, let's print the port there.

In summary, we want to replace all ugly custom PortCfg reporting code with either

* the for-range port loop that automatically reports the current port context,

* the new PortCfg printing operator, or

* the newly added directiveName field.